src/commands/secret/addSecret/SecretValueStep.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep } from "@microsoft/vscode-azext-utils";
+import { localize } from "../../../utils/localize";
+import { validateUtils } from "../../../utils/validateUtils";
+import type { ISecretContext } from "../ISecretContext";
+
+export class SecretValueStep extends AzureWizardPromptStep<ISecretContext> {
+    public async prompt(context: ISecretContext): Promise<void> {
+        context.secretValue = await context.ui.showInputBox({
+            prompt: localize('secretValue', 'Enter a secret value.'),
+            password: true,
+            validateInput: this.validateInput
+        });
+        context.valuesToMask.push(context.secretValue);
+    }
+
+    public shouldPrompt(context: ISecretContext): boolean {
+        return !context.secretValue;
+    }
+
+    private validateInput(val: string | undefined): string | undefined {
+        val = val ? val.trim() : '';
```

Comment:
Should the secret be trimmed, even for validation? A lot of services don't work with a secret that begins or ends with whitespace but IMO that's a bug.